src/libtiled/tilelayer.cpp

```diff
@@ -635,14 +635,21 @@ Layer *TileLayer::mergedWith(Layer *other) const
     return merged;
 }
 
-QRegion TileLayer::computeDiffRegion(const TileLayer *other) const
+QRegion TileLayer::computeDiffRegion(const TileLayer *other, const QRect rect) const
 {
     QRegion ret;
 
     const int dx = other->x() - mX;
     const int dy = other->y() - mY;
-    QRect r = QRect(0, 0, width(), height());
-    r &= QRect(dx, dy, other->width(), other->height());
+
+    QRect r;
+
+    if (rect.isEmpty()) {
+        QRect r = QRect(0, 0, width(), height());
+        r &= QRect(dx, dy, other->width(), other->height());
+    } else {
+        r = rect;
+    }
```

Comment:
Why do you need the `rect` parameter? As far as I can see, `r` just needs to cover both layers and it should be in local coordinates, so this should do it:
```c++
const QRect r = bounds().united(other->bounds()).translated(-position());
```